File: android-core/src/test/java/com/mparticle/external/ApiVisibilityTest.java
package com.mparticle.external;

import com.mparticle.MParticle;
import com.mparticle.Session;

import org.junit.Test;

import java.lang.reflect.Method;
import java.lang.reflect.Modifier;

import static org.junit.Assert.assertEquals;

public class ApiVisibilityTest {

    @Test
    public void testMParticleApiVisibility() throws Exception {
        Method[] mpMethods = MParticle.class.getDeclaredMethods();
        int publicMethodCount = 0;
        for (Method m : mpMethods) {
            if (Modifier.isPublic(m.getModifiers())) {
                publicMethodCount++;
            }
        }
        assertEquals(62, publicMethodCount);
    }

    @Test
    public void testSessionApiVisibility() throws Exception {
        Method[] mpMethods = Session.class.getDeclaredMethods();
        int publicMethodCount = 0;
        for (Method m : mpMethods) {
            if (Modifier.isPublic(m.getModifiers())) {
                publicMethodCount++;
            }
        }
        assertEquals(4, publicMethodCount);
    }
}

File: android-core/src/test/kotlin/com/mparticle/external/ApiVisibilityTest.kt
package com.mparticle.external

import com.mparticle.MParticle
import com.mparticle.Session
import org.junit.Assert
import org.junit.Test
import java.lang.reflect.Modifier

class ApiVisibilityTest {
    @Test
    @Throws(Exception::class)
    fun testMParticleApiVisibility() {
        val mpMethods = MParticle::class.java.declaredMethods
        var publicMethodCount = 0
        for (m in mpMethods) {
            if (Modifier.isPublic(m.modifiers)) {
                publicMethodCount++
            }
        }
        Assert.assertEquals(66, publicMethodCount)
    }

    @Test
    @Throws(Exception::class)
    fun testSessionApiVisibility() {
        val mpMethods = Session::class.java.declaredMethods
        var publicMethodCount = 0
        for (m in mpMethods) {
            if (Modifier.isPublic(m.modifiers)) {
                publicMethodCount++
            }
        }
        Assert.assertEquals(4, publicMethodCount)
    }
}
